R/print.R

```diff
@@ -457,3 +423,77 @@ createRegressionDataTable <- function(x, p.cutoff, caption = NULL, coeff.digits
 
 
 
+#' Computes and inspects the variance inflation factors (VIFs) and throws a warning if they are high
+#' @param x A Regression object
+#' @noRd
+checkVIFAndWarn <- function(x)
+{
+    n.coefs <- length(x$original$coefficients)
+    n.variables <- ncol(x$model)
+    if (n.coefs > 2 & n.variables > 2 & x$missing != "Use partial data (pairwise correlations)")
+    {
+        vifs <- tryCatch(vif(x), error = function(e) NULL)
+        if (!is.null(vifs))
+        {
+            # Check if the GVIF is used and square it
+            if (NCOL(vifs) == 3)
+            {
+                vifs <- vifs[, 3]^2
+                prefix <- "squared Generalized "
+            } else
+                prefix <- ""
+            # Check is the largest calculated VIF is large enough for a warning.
+            max.vif <- max(vifs)
+            if (!is.nan(max.vif) && max.vif >= 4)
```

Comment:
Why is the check for NaN needed? Are some of the entries in vif still okay in this case and we could subset and still produce a warning for the other values?